distributed/diagnostics/nvml.py

```diff
@@ -8,7 +9,16 @@ def _pynvml_handles():
     if handles is None:
         pynvml.nvmlInit()
         count = pynvml.nvmlDeviceGetCount()
-        handles = [pynvml.nvmlDeviceGetHandleByIndex(i) for i in range(count)]
+        cuda_visible_devices = [
```

Comment:
As dask cuda just reorder the devices to change cuda device enumeration.  This is still getting all the devices like what nvml does right?

Reply:
I'm not sure that I undrstand this comment.  My understanding is that `pynvml` doesn't respect the `CUDA_VISIBLE_DEVICES` environment variable, and so we need to handle this manually.